sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -204,6 +207,40 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
   }
 
   /**
+   * Create a partition filter specification.
+   */
+  def visitPartitionFilterSpec(ctx: PartitionSpecContext): Expression = withOrigin(ctx) {
+    val parts = ctx.partitionVal.asScala.map { pVal =>
+      val name = pVal.identifier.getText
+      val operator = Option(pVal.comparisonOperator).map(_.getText)
+      validate(operator.nonEmpty && !operator.get.equals("<=>"),
+        "'<=>' operator is not allowed in partition specification.", ctx)
```

Comment:
`quater` has no operator. So, it will not reach here due to `if (operator.isDefined) {`.
To make it sure, I'll add the testcase.